### ISSUE TITLE
Cannot find module '@angular-devkit/schematics' while running cmd: ng new --collection=@angular/bazel

### ISSUE THREAD

**Issue by vltansky** [angular/angular#35221]
# 🐞 Bug report

### Command (mark with an `x`)

- [x] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....

### Description
Trying to run `ng new  --collection=@angular/bazel`
getting error: Cannot find module '@angular-devkit/schematics' 
@angular/cli 9.0.1
@angular/bazel 9.0.0

![image](https://user-images.githubusercontent.com/5851280/74015510-6ade5100-4999-11ea-8d80-df7907e65661.png)

## 🔬 Minimal Reproduction
`npm i -g @angular/cli @angular/bazel`
`ng new ng new  --collection=@angular/bazel`

## 🔥 Exception or Error
<pre><code>
An unhandled exception occurred: Cannot find module '@angular-devkit/schematics'
Require stack:
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\bazel\src\schematics\ng-new\index.js       
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\cli\node_modules\@angular-devkit\schematics\tools\export-ref.js
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\cli\node_modules\@angular-devkit\schematics\tools\index.js
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\cli\utilities\json-schema.js
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\cli\models\command-runner.js
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\cli\lib\cli\index.js
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\cli\lib\init.js
- C:\Users\vltan\AppData\Roaming\npm\node_modules\@angular\cli\bin\ng
See "C:\Users\vltan\AppData\Local\Temp\ng-FnUFw6\angular-errors.log" for further details.
</code></pre>

## 🌍 Your Environment
<pre><code>
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 9.0.1
Node: 12.15.0
OS: win32 x64

Angular:
...
Ivy Workspace:

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.900.1
@angular-devkit/core         9.0.1
@angular-devkit/schematics   9.0.1
@schematics/angular          9.0.1
@schematics/update           0.900.1
rxjs                         6.5.3
</code></pre>

**Anything else relevant?**

**Comment by blogcraft** [comment]
I have the exactly same issue since RC8

**Comment by blogcraft** [comment]
Installing 
@angular-devkit/schematics 
@angular-devkit/core
@schematics/angular

did the trick, but I guess that this should not need to be a thing in a normal update

**Comment by vltansky** [comment]
Did you install these globally?

**Comment by blogcraft** [comment]
No, installed in the project, then run the update.

Then, after the migrations finished, in the logs it displayed a message indicating that they can be removed and not necessary. (sorry I cannot paste the original message, I already had closed it)

**Comment by tfrijsewijk** [comment]
I had the same error. Fixed it with

```
npm install @angular-devkit/schematics @angular-devkit/core @schematics/angular
```

Then updated with

```
npm run ng update @angular/core @angular/cli
```

Received these messages:

```
    Package "@angular-devkit/schematics" found in the workspace package.json. This package typically does not need to be installed manually. If it is not being used by project code, it can be removed from the package.json.
    Package "@angular-devkit/core" found in the workspace package.json. This package typically does not need to be installed manually. If it is not being used by project code, it can be removed from the package.json.
```

Although it only mentions two of the three newly installed packages, i uninstalled all three of them:

```
npm uninstall @angular-devkit/schematics @angular-devkit/core @schematics/angular
```

**Comment by blogcraft** [comment]
Exactly, that was the message.

**Comment by simeyla** [comment]
> Did you install these globally?
> No, installed in the project, then run the update.

But if you're running `ng new --collection=@angular/bazel` then you don't *have* a project.

I ran them globally. Now I get `Cannot find module 'tslib'`

Seems they weren't kidding about bazel not being production ready for angular.

**Comment by DanielHabenicht** [comment]
Was also a problem for me, https://github.com/angular/angular/issues/35221#issuecomment-584170071 helped.

**Comment by vltansky** [comment]
> > Did you install these globally?
> > No, installed in the project, then run the update.
> 
> But if you're running `ng new --collection=@angular/bazel` then you don't _have_ a project yet!
> 
> I ran them globally. Then I got `Cannot find module 'tslib'`. This worked to fix that:
> 
> ```
> npm install -g tslib
> ```
> 
> I can actually run `ng serve` now for the sample project, but I just cannot get past this:
> 
> > iBazel [3:33PM]: Querying for files to watch...
> > )) was unexpected at this time.
> > iBazel [3:33PM]: Bazel query failed: exit status 255
> 
> Seems they weren't kidding about bazel not being production ready for angular.

Exactly. The issue about creating new project, not updating or `ng add @angular/bazel`. 
Installing modules globally did the trick.
So, just to summarize run: 
1.`npm i -g @angular-devkit/schematics @angular-devkit/core @schematics/angular tslib` 
2. `ng new --collection=@angular/bazel` 
3. `ng serve`
I succeeded to build a project with `ng serve` on `angular 9.0.5`

so the bug is actually about having to manually install modules globally

**Comment by afifalfiano** [comment]
> > > Did you install these globally?
> > > No, installed in the project, then run the update.
> > 
> > 
> > But if you're running `ng new --collection=@angular/bazel` then you don't _have_ a project yet!
> > I ran them globally. Then I got `Cannot find module 'tslib'`. This worked to fix that:
> > ```
> > npm install -g tslib
> > ```
> > 
> > 
> > I can actually run `ng serve` now for the sample project, but I just cannot get past this:
> > > iBazel [3:33PM]: Querying for files to watch...
> > > )) was unexpected at this time.
> > > iBazel [3:33PM]: Bazel query failed: exit status 255
> > 
> > 
> > Seems they weren't kidding about bazel not being production ready for angular.
> 
> Exactly. The issue is about creating **new project**, not updating or `ng add @angular/bazel`.
> Installing modules globally did the trick.
> So, just to summarize:
> **run:**
> 1.`npm i -g @angular-devkit/schematics @angular-devkit/core @schematics/angular tslib`
> 2. `ng new --collection=@angular/bazel`
> 3. `ng serve`
> I succeeded to build a project with `ng serve` on `angular 9.0.5`
> 
> so the bug is actually about having to manually install modules globally

this is work for me. Thanks

**Comment by atiris** [comment]
Test with the latest version angular/bazel@9.1.3
TLDR; Bug is still present

State before:

```bash
$ npm -g list --depth=0
/home/jozef/.npm-global/lib
├── @angular/cli@9.1.3
├── npm@6.14.4
└── typescript@3.8.3
```

Following docs: https://angular.io/guide/bazel#using-bazel-with-the-angular-cli

```bash
$ npm install -g @angular/bazel
$ npm -g list --depth=0
/home/jozef/.npm-global/lib
├── @angular/bazel@9.1.3
├── @angular/cli@9.1.3
├── UNMET PEER DEPENDENCY @angular/compiler-cli@9.1.3
├── UNMET PEER DEPENDENCY @bazel/typescript@>=1.0.0
├── npm@6.14.4
├── UNMET PEER DEPENDENCY rollup@>=1.20.0
├── UNMET PEER DEPENDENCY rollup-plugin-commonjs@>=9.0.0
├── UNMET PEER DEPENDENCY rollup-plugin-node-resolve@>=4.2.0
├── UNMET PEER DEPENDENCY rollup-plugin-sourcemaps@>=0.4.0
└── typescript@3.8.3

npm ERR! peer dep missing: @angular/compiler-cli@9.1.3, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: @bazel/typescript@>=1.0.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup@>=1.20.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup-plugin-commonjs@>=9.0.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup-plugin-node-resolve@>=4.2.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup-plugin-sourcemaps@>=0.4.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: typescript@~3.8.2, required by tsickle@0.38.1
```

New Angular app

```bash
$ ng n test --routing=false --prefix=test --style=css --minimal=true --collection=@angular/bazel
An unhandled exception occurred: Cannot find module '@angular-devkit/schematics'
Require stack:
- /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/export-ref.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/index.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/utilities/json-schema.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/models/command-runner.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/cli/index.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/init.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/bin/ng
See "/tmp/ng-uR7ayE/angular-errors.log" for further details.
```

```bash
$ cat /tmp/ng-uR7ayE/angular-errors.log
[error] Error: Cannot find module '@angular-devkit/schematics'
Require stack:
- /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/export-ref.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/index.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/utilities/json-schema.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/models/command-runner.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/cli/index.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/init.js
- /home/jozef/.npm-global/lib/node_modules/@angular/cli/bin/ng
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:982:15)
    at Function.Module._load (internal/modules/cjs/loader.js:864:27)
    at Module.require (internal/modules/cjs/loader.js:1044:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js:21:24
    at /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js:12:17
    at Object.<anonymous> (/home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js:18:3)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
```

```bash
Angular CLI: 9.1.3
Node: 12.16.1
OS: linux x64

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.901.3
@angular-devkit/core         9.1.3
@angular-devkit/schematics   9.1.3
@schematics/angular          9.1.3
@schematics/update           0.901.3
rxjs                         6.5.4
```

**Comment by afifalfiano** [comment]
> Test with the latest version angular/bazel@9.1.3
> TLDR; Bug is still present
> 
> State before:
> 
> ```shell
> $ npm -g list --depth=0
> /home/jozef/.npm-global/lib
> ├── @angular/cli@9.1.3
> ├── npm@6.14.4
> └── typescript@3.8.3
> ```
> 
> Following docs: https://angular.io/guide/bazel#using-bazel-with-the-angular-cli
> 
> ```shell
> $ npm install -g @angular/bazel
> $ npm -g list --depth=0
> /home/jozef/.npm-global/lib
> ├── @angular/bazel@9.1.3
> ├── @angular/cli@9.1.3
> ├── UNMET PEER DEPENDENCY @angular/compiler-cli@9.1.3
> ├── UNMET PEER DEPENDENCY @bazel/typescript@>=1.0.0
> ├── npm@6.14.4
> ├── UNMET PEER DEPENDENCY rollup@>=1.20.0
> ├── UNMET PEER DEPENDENCY rollup-plugin-commonjs@>=9.0.0
> ├── UNMET PEER DEPENDENCY rollup-plugin-node-resolve@>=4.2.0
> ├── UNMET PEER DEPENDENCY rollup-plugin-sourcemaps@>=0.4.0
> └── typescript@3.8.3
> 
> npm ERR! peer dep missing: @angular/compiler-cli@9.1.3, required by @angular/bazel@9.1.3
> npm ERR! peer dep missing: @bazel/typescript@>=1.0.0, required by @angular/bazel@9.1.3
> npm ERR! peer dep missing: rollup@>=1.20.0, required by @angular/bazel@9.1.3
> npm ERR! peer dep missing: rollup-plugin-commonjs@>=9.0.0, required by @angular/bazel@9.1.3
> npm ERR! peer dep missing: rollup-plugin-node-resolve@>=4.2.0, required by @angular/bazel@9.1.3
> npm ERR! peer dep missing: rollup-plugin-sourcemaps@>=0.4.0, required by @angular/bazel@9.1.3
> npm ERR! peer dep missing: typescript@~3.8.2, required by tsickle@0.38.1
> ```
> 
> New Angular app
> 
> ```shell
> $ ng n test --routing=false --prefix=test --style=css --minimal=true --collection=@angular/bazel
> An unhandled exception occurred: Cannot find module '@angular-devkit/schematics'
> Require stack:
> - /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/export-ref.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/index.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/utilities/json-schema.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/models/command-runner.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/cli/index.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/init.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/bin/ng
> See "/tmp/ng-uR7ayE/angular-errors.log" for further details.
> ```
> 
> ```shell
> $ cat /tmp/ng-uR7ayE/angular-errors.log
> [error] Error: Cannot find module '@angular-devkit/schematics'
> Require stack:
> - /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/export-ref.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/node_modules/@angular-devkit/schematics/tools/index.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/utilities/json-schema.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/models/command-runner.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/cli/index.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/lib/init.js
> - /home/jozef/.npm-global/lib/node_modules/@angular/cli/bin/ng
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:982:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:864:27)
>     at Module.require (internal/modules/cjs/loader.js:1044:19)
>     at require (internal/modules/cjs/helpers.js:77:18)
>     at /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js:21:24
>     at /home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js:12:17
>     at Object.<anonymous> (/home/jozef/.npm-global/lib/node_modules/@angular/bazel/src/schematics/ng-new/index.js:18:3)
>     at Module._compile (internal/modules/cjs/loader.js:1158:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
>     at Module.load (internal/modules/cjs/loader.js:1002:32)
> ```
> 
> ```shell
> Angular CLI: 9.1.3
> Node: 12.16.1
> OS: linux x64
> 
> Package                      Version
> ------------------------------------------------------
> @angular-devkit/architect    0.901.3
> @angular-devkit/core         9.1.3
> @angular-devkit/schematics   9.1.3
> @schematics/angular          9.1.3
> @schematics/update           0.901.3
> rxjs                         6.5.4
> ```
I Think you need to install this dependency, because to installing bazel you need some stuff, Hopefully can be solve 👍
npm ERR! peer dep missing: @angular/compiler-cli@9.1.3, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: @bazel/typescript@>=1.0.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup@>=1.20.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup-plugin-commonjs@>=9.0.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup-plugin-node-resolve@>=4.2.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: rollup-plugin-sourcemaps@>=0.4.0, required by @angular/bazel@9.1.3
npm ERR! peer dep missing: typescript@~3.8.2, required by tsickle@0.38.1

**Comment by jessicajaniuk** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular version.

If the problem still exists in your application, please [open a new issue](https://github.com/angular/angular/issues/new/choose) and follow the instructions in the issue template.
